userdata/userdata_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2023 Steffen Vogel.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package userdata_test
+
+import (
+	"bytes"
+	"encoding/hex"
+	"testing"
+
+	"github.com/google/nftables"
+	"github.com/google/nftables/userdata"
+)
+
+func TestUserDataComment(t *testing.T) {
+	r := nftables.Rule{}
+
+	wantComment := "this is my comment"
+	want := []byte{0, 19, 116, 104, 105, 115, 32, 105, 115, 32, 109, 121, 32, 99, 111, 109, 109, 101, 110, 116, 0}
+
+	r.UserData = userdata.PutString(r.UserData, userdata.TypeComment, wantComment)
+
+	if !bytes.Equal(r.UserData, want) {
+		t.Fatalf("UserData mismatch: %s != %s",
+			hex.EncodeToString(r.UserData),
+			hex.EncodeToString(want))
+	}
+
+	if comment, ok := userdata.GetString(r.UserData, userdata.TypeComment); !ok {
+		t.Fatalf("failed to get comment")
+	} else if comment != wantComment {
+		t.Fatalf("comment does not match: %s != %s", comment, wantComment)
+	}
+}
+
+func TestInt(t *testing.T) {
+	r := nftables.Rule{}
+
+	wantRuleID := uint32(1234)
+	want := []byte{100, 4, 210, 4, 0, 0}
+
+	r.UserData = userdata.PutInt(r.UserData, userdata.TypeRuleID, wantRuleID)
+
+	if !bytes.Equal(r.UserData, want) {
+		t.Fatalf("UserData mismatch: %s != %s",
```

Comment:
Would %x not work here (instead of calling hex.EncodeToString manually)?

Reply:
Thanks, I didnt knew `%x` yet :)